### ISSUE TITLE
[Question] Explanation needed on input and output_chunk_length

### ISSUE THREAD

**Issue by PranayMehta** [unit8co/darts#2513]
**About**
I am trying to implement time series forecasting models using darts. My task is to first train univariate forecasting models and then try to establish a baseline. Once done, then I want to train my models using past co-variates and see if the error can improve. I want to **predict next 90 days** of energy demand using historicals and then using historicals + past co-variates.

**Question**
While coming across documentation in darts, I am slightly confused about how some of parameters are leveraged when training these models.

For example, when using `Linear Regression`, there is a parameter called `output_chunk_length`. How does it affect the output or the training ?

If we consider linear regression with univariate models, then the equation (example) is Yt = Yt-1 + Yt-7 etc. 

How does the model get fit if 
Case A - I pass `output_chunk_length = 7`
Case B - I pass `output_chunk_length = 90`

Can someone explain in layman terms of how the model outputs are affected while training  when this parameter changes? Is something similar followed with input_chunk_length too?

**Comment by eschibli** [comment]
If `multi_models=True` (the default), `output_chunk_length` different, independent linear models are trained, one for each horizon up to `output_chunk_length`.

**Comment by madtoinou** [comment]
Hi @PranayMehta, I would recommend going trough the [Quickstart Notebook](https://unit8co.github.io/darts/quickstart/00-quickstart.html) and the [RegressionModel Example Notebook](https://unit8co.github.io/darts/examples/20-RegressionModel-examples.html) to familiarize yourself with the Darts terminology and features. The impact of `input_chunk_length` and `output_chunk_length` are explained in details :)

**Comment by madtoinou** [comment]
1. When `multi_models=False`, the single model is still responsible for predicting all the steps in the `output_chunk_length`, not only the last one. The timeseries in the dataset are tabularized as pictured in the images in order to obtain the desired outcome (it's not just a sliding window).
2. Since there is only one set of coefficients (when `multi_models=False`, assuming the model is a linear regression for the sake of the example), the inputs need to be different to forecast each step or the models would predict the same value for all the steps in `output_chunk_lengths`. Hence the shift of the lags by `output_chunk_length - n` when forecasting the `n`-th step in `output_chunk_length` and the fact that forecasting of `t0` does not rely on `t-1` (`output_chunk_length - n = 2 - 1 = 1`).
3. By convention, `t0` is always the first step of the forecast horizon.

I hope that it clarifies things.

**Comment by madtoinou** [comment]
Hi @wehrlik,

I understand your point; the gap between the most recent lags and the forecasted step definitely impact the quality of the forecasts. However, in order to have a single model ("one set of coefficients/weights") forecasting all the step in `output_chunk_length` without relying on auto-regression, applying a shift to the lags is a pretty traditional approach. Some alternatives were mentioned in the issue #2234, we are reading about them and trying to see how they could be added to Darts.

In the meantime, if you want to be able to avoid the gap between the lags and the forecasted step, you can simply use `multi_models=True`. And if you want to be able to use the forecast of t0 to forecast t+1, set `output_chunk_length=1`. Each of these approaches has pros & cons, it ultimately boils down to what you want to achieve.

Since the original question has been answered and the discussion is slightly drifting on another topic, I am going to close this issue, feel free to open a new one if anything remains unclear.
